cmd/tcpgoon.go

```diff
@@ -53,26 +57,65 @@ func init() {
 	runCmd.Flags().BoolVarP(&params.debug, "debug", "d", false, "Print debugging information to the standard error")
 	runCmd.Flags().IntVarP(&params.reportingInterval, "interval", "i", 1, "Interval, in seconds, between stats updates")
 	runCmd.Flags().BoolVarP(&params.assumeyes, "assume-yes", "y", false, "Force execution without asking for confirmation")
+	runCmd.Flags().BoolVarP(&params.docker, "docker", "", false, "Running against a docker image. " +
+	"Use the image reference instead of the <host>, and a custom mapped port if the first exposed one is not ok (optional")
 }
 
 func validateRequiredArgs(params *tcpgoonParams, args []string) error {
-	if len(args) != 2 {
-		return errors.New("Number of required parameters doesn't match")
+	var minNumberRequiredArgs int
+	var maxNumberRequiredArgs int
+
+	if params.docker {
+		minNumberRequiredArgs = 1
+		maxNumberRequiredArgs = 2
+	} else {
+		minNumberRequiredArgs = 2
+		maxNumberRequiredArgs = 2
 	}
-	params.target = args[0]
-	addrs, err := net.LookupIP(params.target)
-	if err != nil || len(addrs) == 0 {
-		return errors.New("Domain name not resolvable")
+
+	if len(args) < minNumberRequiredArgs || len(args) > maxNumberRequiredArgs{
+		return errors.New("Number of required parameters doesn't match")
 	}
 
-	params.targetip = addrs[0].String()
-	fmt.Fprintln(debugging.DebugOut, "TCPGOON target: Hostname(", params.target, "), IP (", params.targetip, ")")
+	if params.docker {
```

Comment:
this massive if/else is... :) . That smells to two different validation functions depending on the "mode", but again, we should rethink the parms in general